react/features/web-hid/webhid-manager.ts

```diff
@@ -0,0 +1,1019 @@
+} from './types';
+import {
+    DEVICE_USAGE,
+    TELEPHONY_DEVICE_USAGE_PAGE,
+    requestTelephonyHID
+} from './utils';
+
+/**
+ * WebHID manager that incorporates all hid specific logic.
+ *
+ * @class WebHidManager
+ */
+export default class WebHidManager extends EventTarget {
+    hidSupport: boolean;
+    deviceInfo: IDeviceInfo;
+    availableDevices: HIDDevice[];
+    isParseDescriptorsSuccess: boolean;
+    outputEventGenerators: { [key: string]: Function; };
+    deviceCommand = {
+        outputReport: {
+            mute: {
+                reportId: 0,
+                usageOffset: -1
+            },
+            offHook: {
+                reportId: 0,
+                usageOffset: -1
+            },
+            ring: {
+                reportId: 0,
+                usageOffset: 0
+            },
+            hold: {
+                reportId: 0,
+                usageOffset: 0
+            }
+        },
+        inputReport: {
+            hookSwitch: {
+                reportId: 0,
+                usageOffset: -1,
+                isAbsolute: false
+            },
+            phoneMute: {
+                reportId: 0,
+                usageOffset: -1,
+                isAbsolute: false
+            }
+        }
+    };
+
+    private static instance: WebHidManager;
+
+    /**
+     *  WebHidManager getInstance.
+     *
+     * @static
+     * @returns {WebHidManager}  - WebHidManager instance.
+     */
+    static getInstance(): WebHidManager {
+        if (!this.instance) {
+            this.instance = new WebHidManager();
+        }
+
+        return this.instance;
+    }
+
+    /**
+     * Creates an instance of WebHidManager.
+     *
+     */
+    constructor() {
+        super();
+
+        this.deviceInfo = {} as IDeviceInfo;
+        this.hidSupport = this.isSupported();
+        this.availableDevices = [];
+        this.isParseDescriptorsSuccess = false;
+        this.outputEventGenerators = {};
+    }
+
+    /**
+     * Check support of hid in navigator.
+     * - experimental API in Chrome.
+     *
+     * @returns {boolean} - True if supported, otherwise false.
+     */
+    isSupported(): boolean {
+        // @ts-ignore
+        return !(!window.navigator.hid || !window.navigator.hid.requestDevice);
+    }
+
+    /**
+     * Handler for for requesting telephony hid devices.
+     *
+     * @returns {void}
+     */
+    async requestHidDevices() {
+        if (!this.hidSupport) {
+            logger.warn('The WebHID API is NOT supported!');
+
+            return false;
+        }
+
+        if (this.deviceInfo?.device && this.deviceInfo.device.opened) {
+            await this.close();
+        }
+
+        // @ts-ignore
+        const devices = await navigator.hid.requestDevice(requestTelephonyHID);
+
+        if (!devices || !devices.length) {
+            logger.warn('No HID devices selected.');
+
+            return false;
+        }
+
+        this.availableDevices = devices;
+
+        return true;
```

Comment:
I'd say this function should return the available devices. Sounds natural given it's called "requestSomething"